funcbench/main.go

```diff
@@ -0,0 +1,362 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type gitClient struct {
+	owner    string
+	branch   string
+	repo     string
+	prNumber int
+}
+type gitHubClient struct {
+	owner            string
+	repo             string
+	latestCommitHash string
+	logLink          string
+	prNumber         int
+	client           *github.Client
+	event            *github.IssueCommentEvent
+}
+
+type benchmarkTester struct {
+	bechRegex                string
+	home                     string
+	comparisonBranchFilePath string
+	pullRequestFilePath      string
+	raceFlagEnabled          bool
+}
+
+func main() {
+	app := kingpin.New(filepath.Base(os.Args[0]), "benchmark result posting and formating tool.\n-i location of github hook file (even.json)")
+	app.HelpFlag.Short('h')
+	input := app.Flag("input", "path to event.json").Short('i').Default("/github/workflow/event.json").String()
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	data, err := ioutil.ReadFile(*input)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// Temporary fix for the new Github actions time format, as go-github is not capable of parsing it. This makes the time stamps unusable.
+	txt := string(data)
+	reg := regexp.MustCompile("(.*)\"[0-9]+/[0-9]+/2019 [0-9]+:[0-9]+:[0-9]+ [AP]M(.*)")
+	txt = reg.ReplaceAllString(txt, "$1\"2019-06-11T09:26:28Z$2")
+	data = []byte(txt)
+	// End of the temporary fix.
+
+	event, err := github.ParseWebHook("issue_comment", data)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	switch eventType := event.(type) {
+	case *github.IssueCommentEvent:
+		ghClient := newGitHubClient(eventType)
+		gitClient, err := newGitClient(eventType)
+		benchTest, err := newBenchmarkTester()
+
+		// Environment setup.
+		if err := os.Setenv("GO111MODULE", "on"); err != nil {
+			log.Fatalln(err)
+		}
+		if err := os.Chdir(os.Getenv("GITHUB_WORKSPACE")); err != nil {
+			log.Fatalln(err)
+		}
+
+		if err := gitClient.cloneRepository(); err != nil {
+			log.Fatalln(err)
+		}
+
+		if err := os.Chdir(ghClient.repo); err != nil {
+			log.Fatalln(err)
+		}
+		if err := setupGitConfig(); err != nil {
+			log.Fatalln(err)
+		}
+
+		if err := gitClient.fetchPullRequest(); err != nil {
+			log.Fatalln(err)
+		}
+		if err := gitClient.checkoutToPullRequest(); err != nil {
+			comment := fmt.Sprintf("Switch to a pull request branch failed. %s", ghClient.logLink)
+			if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+				log.Fatalf("An error: %v occured while processing error: %v", postCommentErr, err)
```

Comment:
```suggestion
				log.Fatalf("error posting a comment for `checkoutToPullRequest` command execution error. checkoutToPullRequest err:%v, postComment err:%v", err,postCommentErr)
```

the same for the one below